src/components/MultiSelect/MultiSelect.test.tsx

```diff
@@ -0,0 +1,744 @@
+        label="MultiSelect Label"
+        variant="dialog"
+        items={items}
+        selectedItems={selectedTestItems}
+        onChange={onChangeMock}
+        onMixedStateChange={onMixedStateChangeMock}
+        onClear={() => null}
+        onApply={() => null}
+      />
+    );
+    // Open menu
+    userEvent.click(
+      screen.getByRole("button", {
+        name: /multiselect label/i,
+      })
+    );
+
+    expect(
+      screen.getByRole("checkbox", { name: /colors/i })
+    ).toBeInTheDocument();
+    userEvent.click(screen.getByRole("checkbox", { name: /colors/i }));
+    expect(onMixedStateChangeMock).toBeCalledTimes(1);
+    expect(onChangeMock).not.toBeCalled();
+
+    userEvent.click(screen.getByRole("checkbox", { name: /colors/i }));
+    expect(onMixedStateChangeMock).toBeCalledTimes(2);
+    expect(onChangeMock).not.toBeCalled();
+  });
+
+  it("should have indeterminate state for parent item if not all child items are checked", () => {
+    render(<MultiSelectTestDialogComponent id="multiselect-dialog-test-id" />);
+    // Open menu
+    userEvent.click(screen.getByRole("button", { name: /MultiSelect Label/i }));
+    // Check the child
+    userEvent.click(screen.getByText("Red"));
+    // Child is checked
+    expect(screen.getByLabelText("Red")).toBeChecked();
+    // Parent is indeterminated
+    expect(screen.getByLabelText("Colors")).toBePartiallyChecked();
+  });
+
+  it("should check all child items if parent is checked", () => {
+    render(<MultiSelectTestDialogComponent id="multiselect-dialog-test-id" />);
+    // Open menu
+    userEvent.click(screen.getByRole("button", { name: /MultiSelect Label/i }));
+    // Check the parent item
+    userEvent.click(screen.getByText("Colors"));
+    // Parent is checked
+    expect(screen.getByLabelText("Colors")).toBeChecked();
+    // Children are checked
+    expect(screen.getByLabelText("Red")).toBeChecked();
+    expect(screen.getByLabelText("Blue")).toBeChecked();
+  });
+
+  it("should check parent item if all child items are checked", () => {
+    selectedTestItems = {
+      "multiselect-dialog-test-id": { items: ["red", "blue"] },
+    };
+
+    render(
+      <MultiSelect
+        id="multiselect-dialog-test-id"
+        label="MultiSelect Label"
+        variant="dialog"
+        items={items}
+        isDefaultOpen={true}
+        selectedItems={selectedTestItems}
+        onMixedStateChange={() => null}
+        onChange={() => null}
+        onClear={() => null}
+        onApply={() => null}
+      />
+    );
+    expect(screen.getByLabelText("Colors")).toBeChecked();
+  });
+  it("should render a count button with the correct count, should clear the selectedItems on click ", () => {
+    const { container } = render(
+      <MultiSelectTestDialogComponent id="multiselect-dialog-test-id" />
+    );
+    // Check for the selectedItems count button to not be present
+    expect(container.querySelector(countButton)).not.toBeInTheDocument();
```

Comment:
What I meant is that `countButton` should be `container.querySelector("span[role='button']")` instead of just `"span[role='button']"`. Did you try this and it didn't work? If it works, make this update and you can put the variable for this in this "it" block and then every other "it" should have it's own variable as well.

Reply:
This has been changed. Note: the first time we check for the count button the button is not there yet. Therefor it is now kept as the initial query (it returns null). After a item has been selected I add the variable and can use it from there on for the whole "it" block.